000X-B-C-I-R-S-error-handling-codes.md

```diff
@@ -0,0 +1,307 @@
+  // both explain the error and offer an actionable resolution to it.
+  string message = 2;
+
+  // Additional error information that the client code can use to handle
+  // the error, such as retry info or a help link.
+  repeated google.protobuf.Any details = 3;
+}
+```
+
+Here is one of the possible details that can be added to the above
+error structure. This is defined in the [error_details.proto Protobuf](https://github.com/googleapis/googleapis/blob/master/google/rpc/error_details.proto) 
+
+```go
+message ErrorInfo {
+  // The reason of the error. This is a constant value that identifies the
+  // proximate cause of the error. Error reasons are unique within a particular
+  // domain of errors. This should be at most 63 characters and match a
+  // regular expression of `[A-Z][A-Z0-9_]+[A-Z0-9]`, which represents
+  // UPPER_SNAKE_CASE.
+  string reason = 1;
+
+  // The logical grouping to which the "reason" belongs. The error domain
+  // is typically the registered service name of the tool or product that
+  // generates the error. Example: "pubsub.googleapis.com". If the error is
+  // generated by some common infrastructure, the error domain must be a
+  // globally unique value that identifies the infrastructure. For Google API
+  // infrastructure, the error domain is "googleapis.com".
+  string domain = 2;
+
+  // Additional structured details about this error.
+  //
+  // Keys should match /[a-zA-Z0-9-_]/ and be limited to 64 characters in
+  // length. When identifying the current value of an exceeded limit, the units
+  // should be contained in the key, not the value.  For example, rather than
+  // {"instanceLimit": "100/request"}, should be returned as,
+  // {"instanceLimitPerRequest": "100"}, if the client exceeds the number of
+  // instances that can be created in a single (batch) request.
+  map<string, string> metadata = 3;
+}
+```
+
+### Error Status
+The properties of the **google.rpc.Status** will be populated as following:
+
+- **Code**  - Protocol level error code. These could be either gRPC or HTTP error codes. See (gRPC Codes ProtoBuf)[https://github.com/googleapis/googleapis/blob/master/google/rpc/code.proto]
+   
+   Example: "InvalidArgument Code = 3", "Internal Code = 13"
+
+- **Message**  - Error message.
+- **Details**  - A set of standard error payloads for error details. These list can be found in [Error Details](https://github.com/googleapis/googleapis/blob/master/google/rpc/error_details.proto)
+   
+   Example: "ErrorInfo", "ResourceInfo"
+
+
+
+Below is partial table of the Standard Error code provided by gRPC and how they map to HTTP error codes. The entire list can found in the following links:
+- [Google API Error Handling]https://cloud.google.com/apis/design/errors#handling_errors
+- (gRPC Codes ProtoBuf)[https://github.com/googleapis/googleapis/blob/master/google/rpc/code.proto]
+
+
+|HTTP | gRPC      | Description |
+|---- | ----------- | ----------- |
+|200  |	OK		  | No error. |
```

Comment:
Probably list all the current set of HTTP return codes and their mappings that dapr currently returns.

Reply:
At the moment, we don't have a mapping to Dapr specific errors. As we discussed in the review call, we should know more as we dig deeper.